pkg/pfcpsim/pfcpsim.go

```diff
@@ -180,11 +180,22 @@ func (c *PFCPClient) PeekNextHeartbeatResponse() (*message.HeartbeatResponse, er
 // It's a blocking operation, which is timed out after c.responseTimeout period (5 seconds by default).
 // Use SetPFCPResponseTimeout() to configure a custom timeout.
 func (c *PFCPClient) PeekNextResponse() (message.Message, error) {
-	select {
-	case msg := <-c.recvChan:
-		return msg, nil
-	case <-time.After(c.responseTimeout):
-		return nil, NewTimeoutExpiredError()
+	var resMsg message.Message = nil
+	var err error = nil
+	delay := time.NewTimer(c.responseTimeout)
+	for {
```

Comment:
Do we need for loop here? It wasn't used before.

Reply:
@osinstom 
In the case of the pfcp message received, it only change the value of resMsg but not return.
At mean while, it also checks the status of the timer, if the timer still running then send the event to channel for terminating the timer.
After that, handler will going to the select timer case, and check whether resMeg exists.

Reply:
Hi @osinstom 
Please help to review the reply when you feel free.
Thanks!